app/src/main/java/com/example/flickrgallery/data/source/StoredLocationsRoomDataSource.kt

```diff
@@ -0,0 +1,55 @@
+package com.example.flickrgallery.data.source
+
+import com.example.data.source.StoredLocationsDataSource
+import com.example.domain.StoredLocation as DomainStoredLocation
+import com.example.flickrgallery.db.Db
+import com.example.flickrgallery.db.StoredLocationDao
+import com.example.flickrgallery.model.StoredLocation as FrameworkStoredLocation
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.flow
+
+class StoredLocationsRoomDataSource (database: Db) : StoredLocationsDataSource {
+
+    private var storedLocationDao: StoredLocationDao = database.storedLocationDao()
+
+    override suspend fun getAll(): Flow<List<DomainStoredLocation>> = flow {
+        storedLocationDao.getAllFlow().collect { frameworkStoredLocations ->
+            emit(frameworkStoredLocations.map { it.toDomainStoredLocation() })
+        }
+    }
+
+    override suspend fun insert(storedLocation: DomainStoredLocation) {
+        storedLocationDao.insert(storedLocation.toFrameworkStoredLocation())
+    }
+
+    override suspend fun update(storedLocation: DomainStoredLocation) {
+        storedLocationDao.update(storedLocation.toFrameworkStoredLocation())
+    }
+
+    override suspend fun delete(storedLocation: DomainStoredLocation) {
+        storedLocationDao.delete(storedLocation.toFrameworkStoredLocation())
+    }
+}
+
+fun FrameworkStoredLocation.toDomainStoredLocation():DomainStoredLocation {
+    val frameworkStoredLocation = this
+    return DomainStoredLocation().apply {
```

Comment:
Quizá estaría más claro con un data class y pasarselo todo en el constructor. Por ejemplo: 
DomainStoredLocation(
id = this.id,
latitude = this.latitude
...
)